### PR TITLE
PR: Create an initial package of InputSystem

### PR DESCRIPTION
**Pull request by joelpryde**
Tasks:
- Move all code into Packages/com.unity.inputsystem
- Generate package.json and move over core package template files
- Export docs from GitHub wiki as markdown and add to Packages/com.unity.inputsystem/Documentation
- Fix asmdef files to conform to new package naming
- Add missing individual asmdef files for Extras
- Fix .gitignore/.npmignore files
- Give access to core assembly for tests with InternalsVisibleTo attribute

### CODE REVIEW

File: Packages/com.unity.inputsystem/Extras/DualShock/Unity.InputSystem.DualShock.asmdef [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    "name": "Unity.InputSystem.DualShock",
+    "references": [
+        "Unity.InputSystem"
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/Extras/DualShock/Unity.InputSystem.DualShock.asmdef:4]: Was wondering... should the DLLs be named UnityEngine.XXX or Unity.XXX? The thing is that nothing in this package goes in the new Unity namespace and likely nothing ever will. At this point it seems there will be a new package which adds an ECS-compatible system in the Unity namespace. Seems more fitting to name that one Unity.Input or something and use UnityEngine.InputSystem for this one. Not sure what the convention is, though.

File: Packages/com.unity.inputsystem/InputSystem/AssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1,3 @@
+using System.Runtime.CompilerServices;
+
+[assembly: InternalsVisibleTo("UnityEngine.Experimental.XR.Interaction.Tests")]
```
Comment (Rene-Damm) [inline Packages/com.unity.inputsystem/InputSystem/AssemblyInfo.cs]: Please update the attribute in InputSystem.cs instead.

**Review by Rene-Damm — CHANGES_REQUESTED** [review]
Minor thing about InternalsVisibleTo. Other than that, looks great to me.

File: Packages/com.unity.inputsystem/Extras/DualShock/Unity.InputSystem.DualShock.asmdef (hunk 1 above)
Comment (joelpryde) [reply Packages/com.unity.inputsystem/Extras/DualShock/Unity.InputSystem.DualShock.asmdef:4]: According to the package template it is Unity.XXX (I just changed my package).  I think this is a new change going forward from the scripting team to move everything into Unity.XXX namespace eventually.
https://gitlab.internal.unity3d.com/upm-packages/upm-package-template/blob/master/Tests/Runtime/Unity.YourPackageName.Tests.asmdef

**Review by Rene-Damm — APPROVED** [review]

**Pull request merged**